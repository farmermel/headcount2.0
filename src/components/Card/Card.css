.card {
  width: 200px;
  margin: 20px 10px;
  border: #000 1px solid;
}

h3 {
  font-family: 'Roboto', sans-serif;
}

.above {
  color: #4e7fe0;
}

.below {
  color: #b51e35;
}