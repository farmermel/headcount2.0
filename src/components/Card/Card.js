import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';

const Card = ({ location, data, compare }) => {
  const dataArray = Object.keys(data).map(year => {
    const klassName = data[year] > 0.5 ? 'above' : 'below';

    return (
      <li key={year} 
          className={klassName}>
        {year}: {data[year]}
      </li>
    );
  });

  return (
    <div className='card'
         onClick={() => compare(location)}>
      <h3>{location}</h3>
      <ul>{dataArray}</ul>
    </div>
  );
};

Card.propTypes = {
  location: PropTypes.string,
  data: PropTypes.objectOf(PropTypes.number)
};

export default Card;
