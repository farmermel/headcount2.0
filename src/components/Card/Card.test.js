import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import { shallow, mount } from 'enzyme';

describe('Card', () => {
  let wrapper;
  const mockData = { 2014: 0.4,
      2015: 0.8 };

  beforeEach( () => {
    wrapper = shallow(<Card location='Montana'
                            data={ mockData } />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should receive className based on value of data above/below 0.5', () => {
    expect(wrapper.find('.above').length).toEqual(1);
    expect(wrapper.find('.below').length).toEqual(1);
    //is there a not dumb way to be explicit about the values
  })
})