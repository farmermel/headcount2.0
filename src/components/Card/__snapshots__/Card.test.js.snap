// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
        Object {
            "2014": 0.4,
            "2015": 0.8,
          }
    }
    location="Montana"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Montana
</h3>,
        <ul>
          <li
                    className="below"
          >
                    2014
                    : 
                    0.4
          </li>
          <li
                    className="above"
          >
                    2015
                    : 
                    0.8
          </li>
</ul>,
      ],
      "className": "card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Montana",
        },
        "ref": null,
        "rendered": "Montana",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="below"
>
              2014
              : 
              0.4
</li>,
            <li
              className="above"
>
              2015
              : 
              0.8
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "2014",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2014",
                ": ",
                0.4,
              ],
              "className": "below",
            },
            "ref": null,
            "rendered": Array [
              "2014",
              ": ",
              0.4,
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2015",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2015",
                ": ",
                0.8,
              ],
              "className": "above",
            },
            "ref": null,
            "rendered": Array [
              "2015",
              ": ",
              0.8,
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Montana
</h3>,
          <ul>
            <li
                        className="below"
            >
                        2014
                        : 
                        0.4
            </li>
            <li
                        className="above"
            >
                        2015
                        : 
                        0.8
            </li>
</ul>,
        ],
        "className": "card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Montana",
          },
          "ref": null,
          "rendered": "Montana",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="below"
>
                2014
                : 
                0.4
</li>,
              <li
                className="above"
>
                2015
                : 
                0.8
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "2014",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2014",
                  ": ",
                  0.4,
                ],
                "className": "below",
              },
              "ref": null,
              "rendered": Array [
                "2014",
                ": ",
                0.4,
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2015",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2015",
                  ": ",
                  0.8,
                ],
                "className": "above",
              },
              "ref": null,
              "rendered": Array [
                "2015",
                ": ",
                0.8,
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
